pkg/deploy/util/util.go

```diff
@@ -280,6 +283,35 @@ func GetAvailablePods(pods []api.Pod, minReadySeconds int32) int32 {
 	return available
 }
 
+type updateConfigFunc func(d *deployapi.DeploymentConfig)
+
+// UpdateConfigWithRetries will try to update a deployment config and ignore any update conflicts.
+func UpdateConfigWithRetries(dn client.DeploymentConfigsNamespacer, namespace, name string, applyUpdate updateConfigFunc) (*deployapi.DeploymentConfig, error) {
+	var (
+		config *deployapi.DeploymentConfig
+		err    error
+	)
+	deploymentConfigs := dn.DeploymentConfigs(namespace)
+	resultErr := wait.Poll(10*time.Millisecond, 1*time.Minute, func() (bool, error) {
```

Comment:
updated - I also moved the utility out of deploy/util so that we won't depend on the client library in our utilities